android/java/org/chromium/chrome/browser/ntp/BraveNewTabPageLayout.java

```diff
@@ -1316,92 +616,407 @@ public void run() {
-                        mFirstVisibleCard = linearLayoutManager.findFirstVisibleItemPosition();
-                        mParentScrollView.scrollBy(0, offset + 2);
-                    } catch (Exception e) {
-                        Log.e("bn", "Exception onScrolled:" + e);
+            @Override
+            public void onScrolled(@NonNull RecyclerView recyclerView, int dx, int dy) {
+                super.onScrolled(recyclerView, dx, dy);
+
+                if (mIsDisplayNews) {
+                    int lastVisibleItemPosition =
+                            linearLayoutManager.findLastCompletelyVisibleItemPosition();
+
+                    if (!mNtpAdapter.shouldDisplayNewsLoading()
+                            && lastVisibleItemPosition > mNtpAdapter.getStatsCount()
+                                            + mNtpAdapter.getTopSitesCount()
+                                            + mNtpAdapter.getNewContentCount()) {
+                        if (mNewsSettingsBar.getVisibility() != View.VISIBLE) {
+                            mNewsSettingsBar.setVisibility(View.VISIBLE);
+                        }
+                        mNtpAdapter.setImageCreditAlpha(0f);
+                    } else if (lastVisibleItemPosition > -1) {
+                        if (mNewsSettingsBar.getVisibility() != View.GONE) {
+                            mNewsSettingsBar.setVisibility(View.GONE);
+                        }
+                        mNtpAdapter.setImageCreditAlpha(1f);
+                    }
+
+                    if (mNtpAdapter.isNewContent()) {
+                        int firstVisibleItemPosition =
+                                linearLayoutManager.findFirstVisibleItemPosition();
+
+                        if (firstVisibleItemPosition
+                                >= mNtpAdapter.getStatsCount() + mNtpAdapter.getTopSitesCount()) {
+                            mNewContentLayout.setVisibility(View.VISIBLE);
+                        } else {
+                            mNewContentLayout.setVisibility(View.GONE);
+                        }
+                    } else {
+                        mNewContentLayout.setVisibility(View.GONE);
+                    }
+                } else if (mIsDisplayNewsOptin) {
+                    int lastVisibleItemPosition =
+                            linearLayoutManager.findLastCompletelyVisibleItemPosition();
+
+                    if (lastVisibleItemPosition == mNtpAdapter.getItemCount() - 1) {
+                        mNtpAdapter.setImageCreditAlpha(0f);
+                    } else {
+                        mNtpAdapter.setImageCreditAlpha(1f);
                     }
                 }
-            });
+            }
+        });
+    }
+
+    private void keepPosition() {
+        if (BraveActivity.getBraveActivity() != null
+                && BraveActivity.getBraveActivity().getActivityTab() != null) {
+            Tab tab = BraveActivity.getBraveActivity().getActivityTab();
```

Comment:
we could move that before the condition check and make `if (BraveActivity.getBraveActivity() != null && tab != null)`